File: implementations/dotnet/EmailInquire/Validators/OneTimeProvider.cs
using System.Linq;

namespace EmailInquire.Validators
{
    /// <summary>
    ///     Validation rejecting the one-time providers domains
    /// </summary>
    public class OneTimeProvider : ValidatorBase<OneTimeProvider>
    {
        public override Response Validate()
        {
            return OneTimeProviders?.Contains(Domain) ?? false ? Response.Invalid(Email) : Response.Undefined;
        }
    }
}

File: implementations/ruby/lib/email_inquire/validator/one_time_provider.rb
# frozen_string_literal: true

require "email_inquire/validator/base"

module EmailInquire
  module Validator
    class OneTimeProvider < Base

      DOMAINS = load_data("one_time_providers").freeze

      def validate
        response.invalid! if DOMAINS.include?(domain)
      end

    end
  end
end
